packages/flutter/test/widgets/two_dimensional_viewport_test.dart

```diff
@@ -2713,6 +2713,27 @@ void main() {
       });
     });
 
+    testWidgets('Does not throw when no child is laid out',
+        (WidgetTester tester) async {
+      final TwoDimensionalChildBuilderDelegate delegate =
+          TwoDimensionalChildBuilderDelegate(
+              maxXIndex: 50,
+              maxYIndex: 50,
+              addAutomaticKeepAlives: false,
+              addRepaintBoundaries: false,
+              builder: (BuildContext context, ChildVicinity vicinity) {
+                if (vicinity.xIndex > 10) {
+                  return SizedBox.square(dimension: 200);
+                }
+                return null;
+              });
+      addTearDown(delegate.dispose);
+
+      await tester.pumpWidget(simpleBuilderTest(delegate: delegate));
+      // In case of exception during layout, expect is not reached
+      expect(true, isTrue);
```

Comment:
Could you do `expect(tester.takeException(), isNull);` instead? That way we assert that the layout didn't throw the exception because of the unexpected null value.

Reply:
Awesome! Didn't know we have such API.